pkg/kf/apps/push.go

```diff
@@ -156,13 +175,40 @@ func setupRoutes(cfg pushConfig, appName string, r []v1alpha1.RouteSpecFields) (
 	}
 }
 
+func noCfgScaling(cfg pushConfig) bool {
+	return cfg.MinScale == nil && cfg.MaxScale == nil
+}
+
+func noScaling(app *v1alpha1.App) bool {
+	return app.Spec.Instances.Exactly == nil &&
+		app.Spec.Instances.Min == nil &&
+		app.Spec.Instances.Max == nil
+}
+
 func mergeApps(cfg pushConfig, hasDefaultRoutes bool) func(newapp, oldapp *v1alpha1.App) *v1alpha1.App {
 	return func(newapp, oldapp *v1alpha1.App) *v1alpha1.App {
 
 		if len(oldapp.Spec.Routes) > 0 && hasDefaultRoutes {
 			newapp.Spec.Routes = oldapp.Spec.Routes
 		}
 
+		// Scaling overrides
+		if noCfgScaling(cfg) {
```

Comment:
Is this the way it works in cf? If not, I'd like to just scale back to 1 by default every time. That makes the push model easier to reason about.

Reply:
cf won't mess with your scaling unless you tell it to:

```
$ cf scale myapp -i 3
$ cf push myapp
$ cf scale myapp
...
instances: 3
```